Lpc compute docs (#2651)
buildkite/docs: pages/buildkite_compute/overview.md

> # What is compute

Reviewer: ```suggestion
# What are hosted agents
```

> Buildkite compute provides a fully managed platform for you to run your agents.

Reviewer: ```suggestion
Buildkite hosted agents provides a fully-managed platform on which you can run your agents.
```

> With Compute Services, the infrastructure management tasks traditionally handled by your team, such as provisioning, scaling, and maintaining the servers that run your agents, can now be managed by Buildkite.

Reviewer: ```suggestion
With hosted agents, the infrastructure management tasks traditionally handled by your team, such as provisioning, scaling, and maintaining the servers that run your agents, can now be managed by Buildkite.
```

> Buildkite compute is currently in private trials, you need to contact support to express interest and have the service switched on for your organization.

Reviewer: How is this interest expressed to support? Would customers just request support of this feature or is there an initial vetting process required between customers expressing an interest and this feature being activated for them?

```suggestion
Buildkite compute is currently in private trials. You'll need to contact support to express interest and have this service activated for your organization.
```

> Buildkite compute is currently in private trials. You'll need to contact support to express interest and have this service activated for your organization.

Reviewer: ```suggestion
Buildkite hosted agents is currently in private trials. You'll need to contact support to express interest and have this service activated for your organization.
```

> ## Compute types

Reviewer: ```suggestion
## Hosted agent types
```

> During our private trial phase, we are offering both Mac and Linux agents. We plan to extend our services to include Windows agents by late 2024 as part of our ongoing commitment to providing a comprehensive range of options.

Reviewer: Should this be "Mac" or "macOS"?

> During our private trial phase, we are offering both Mac and Linux agents. We plan to extend our services to include Windows agents by late 2024 as part of our ongoing commitment to providing a comprehensive range of options.

Reviewer: ```suggestion
During our private trial phase, Buildkite is offering both Mac and Linux agents. Buildkite plans to extend these services to include Windows agents by late 2024 as part of our ongoing commitment to providing a comprehensive range of options.
```

Should this be "Mac" or "macOS"?

For detailed information about available agent sizes and configuration please see [Mac Compute Instances](/docs/buildkite-compute/macos-instances), and [Linux Compute Instances](/docs/buildkite-compute/linux-instances)

Usage of all instance types is billed on a per-minute basis.

> Every Buildkite hosted agent within a cluster benefits from hypervisor-level isolation, ensuring robust separation between each instance.

Reviewer: ```suggestion
Every Buildkite-hosted agent within a cluster benefits from hypervisor-level isolation, ensuring robust separation between each instance.
```

> ## Creating a compute queue

Reviewer: ```suggestion
## Creating a hosted agent queue
```

> You can set up distinct compute queues, each configured with specific types and sizes to efficiently manage jobs with varying requirements.

Reviewer: ```suggestion
You can set up distinct hosted agent queues, each configured with specific types and sizes to efficiently manage jobs with varying requirements.
```

> For example you may have two queues setup

Reviewer: Missing colon preceding a list and 'set up' is two separate words when this term is used as a verb, [like the others here](https://github.com/buildkite/docs/blob/main/styleguides/writing-style.md#glossary-of-notable-terms-and-their-spelling).

```suggestion
For example you may have two queues set up:
```

* `mac_small_7gb`
* `mac_large_32gb`

> Learn more about best practice queue configuration [here](/docs/clusters/overview#clusters-and-queues-best-practice-how-should-i-structure-my-queues)

Reviewer: As a general rule, probably best to avoid creating links against non-descriptive words like 'here' and instead, generate against the core descriptive text/words itself.

```suggestion
Learn more about [best practices for configuring queues](/docs/clusters/overview#clusters-and-queues-best-practice-how-should-i-structure-my-queues).
```

> To create a compute queue, navigate to the cluster where you want your compute queue to reside, select _New Queue_ and select _Hosted_ as the compute option. follow the prompts to configure your compute services.

Reviewer: ```suggestion
To create a compute queue, navigate to the cluster where you want your compute queue to reside, select _New Queue_ and select _Hosted_ as the compute option. Follow the prompts to configure your compute services.
```

> ## Using GitHub repositories in your compute pipelines

Reviewer: ```suggestion
## Using GitHub repositories in your hosted agent pipelines
```

> Buildkite compute services support both public and private repositories, see [Compute Source Control](/docs/buildkite-compute/source-control) to learn more about setting up code access.

Reviewer: ```suggestion
Buildkite compute services support both public and private repositories. Learn more about setting up code access in [Compute Source Control](/docs/buildkite-compute/source-control).
```

## Migrating your pipelines to compute services

> You can migrate existing pipelines to compute services, to learn more see [Compute Pipeline Migration](/docs/buildkite-compute/pipeline-migration)

Reviewer: ```suggestion
Learn more about migrating existing pipelines to Buildkite compute services in [Compute pipeline migration](/docs/buildkite-compute/pipeline-migration).
```
Is this Buildkite compute or just compute?

## Coming soon

### API support for hosted queues

> We are working on adding functionality in the API to allow configuration of hosted queues.

Reviewer: Try to avoid using 'we' in the docs—it's not always obvious who 'we' is.

```suggestion
Buildkite is working on adding functionality in the API to allow configuration of hosted queues.
```

> ### macOS image configuration in the UI

Reviewer: [Add a line break after the heading](https://github.com/buildkite/docs/blob/main/styleguides/markdown-syntax-style.md#headings) - refer to note at the end of this section.

> We are building the ability to choose the software versions you require to be installed on the MacOS instances used in your queues.

Reviewer: ```suggestion
Buildkite is building the ability to choose the software versions you require to be installed on the macOS instances used in your queues.
```

### Buildkite secrets

> Only in the rarest cases does CI not need to access outside services, and in these cases, the usability of the CI is severely limited. To use CI effectively - and to move toward CD, continuous deployment - your CI system needs to be able to safely and securely interact with outside services like observability platforms, cloud providers, and other services.

Reviewer: Refer to the [guidance on using m-dashes](https://developers.google.com/style/dashes) (from the Google developer styleguide docs).

```suggestion
Only in the rarest cases does CI not need to access outside services, and in these cases, the usability of the CI is severely limited. To use CI effectively—and to move toward CD, continuous deployment—your CI system needs to be able to safely and securely interact with outside services like observability platforms, cloud providers, and other services.
```

> To do this, you need to be able to securely store secrets like API credentials, SSH keys, and other sensitive information, and be able to use them safely and effectively in your builds. Buildkite Secrets provides a way to do this - we'll securely store your secrets, and provide a way for you to access them in your builds.

Reviewer: After this long list of items, be clearer about what the 'them' refers to.

```suggestion
To do this, you need to be able to securely store secrets like API credentials, SSH keys, and other sensitive information, and be able to use these secrets safely and effectively in your builds. Buildkite secrets provides a way to do this—Buildkite securely stores your secrets and provides a way for you to access them in your builds.
```

> Buildkite Secrets are an encrypted key-value store, where secrets are available to your builds via the Buildkite Agent. Secrets are encrypted both at rest and in transit using SSL, and are decrypted server-side when accessed by the agent. The agent makes it easy to use these secrets in your build scripts, and provides a way to inject secrets into your build steps as environment variables.

Reviewer: Would it be SSL or TLS?

```suggestion
Buildkite secrets are an encrypted key-value store, where secrets are available to your builds via the Buildkite agent. Secrets are encrypted both at rest and in transit using SSL, and are decrypted server-side when accessed by the agent. The agent makes it easy to use these secrets in your build scripts, and provides a way to inject secrets into your build steps as environment variables.
```

> Secrets will initially be scoped per-cluster - that is, agents outside of the cluster the secret belongs to will not be able to access that secret. We have additional work on the roadmap to allow secrets to be scoped per-pipeline.

Reviewer: ```suggestion
Secrets will initially be scoped per-cluster—that is, agents not associated with cluster a secret belongs to will not be able to access that secret. Buildkite has additional work on the roadmap to allow secrets to be scoped per-pipeline.
```

> Until Buildkite secrets are available and if you would like to continue using your third party secrets provider like AWS SSM, GC Secrets or Hashicorp Vault we provide plugins that allow you to access these services. If a plugin for the service you use is not listed below please contact support.

Reviewer: Adding a link to our existing docs for the relevant feature described.

```suggestion
Until Buildkite secrets is available and if you would like to continue [using your third party secrets provider like AWS SSM, GC Secrets or Hashicorp Vault](/docs/pipelines/secrets), Buildkite provides plugins that allow you to access these services. If a plugin for the service you use is not listed below please contact support.
```

<table>
    <thead>
        <tr><th>Service</th><th>Plugin</th></tr>
    </thead>
    <tbody>
        <tr><td>AWS SSM</td><td><a href="https://github.com/buildkite-plugins/aws-assume-role-with-web-identity-buildkite-plugin">aws-assume-role-with-web-identity-buildkite-plugin</a></td></tr>
        <tr><td>GC Secrets</td><td><a href="https://github.com/buildkite-plugins/gcp-workload-identity-federation-buildkite-plugin">gcp-workload-identity-federation-buildkite-plugin</a></td></tr>
        <tr><td>Hashicorp Vault</td><td><a href="https://github.com/buildkite-plugins/vault-secrets-buildkite-plugin">vault-secrets-buildkite-plugin</a></td></tr>
    </tbody>
</table>

### Ability to SSH into a machine

> We are working on allowing direct SSH access into the compute instances.

Reviewer: ```suggestion
Buildkite is working on allowing direct SSH access into the compute instances.
```

### Usage metrics

> Enhanced usage metrics across your compute queues.
> 
> 
> 

Reviewer: ```suggestion
Enhanced usage metrics across your hosted agents' queues.
```

What is the purpose of this section? It currently doesn't seem to serve a purpose...
